kiwi/volume_manager/btrfs.py

```diff
@@ -287,6 +287,21 @@ def umount_volumes(self):
 
         return all_volumes_umounted
 
+    def get_mountpoint(self):
+        """
+        Provides btrfs root mount point directory
+
+        Effective use of the directory is guaranteed only after sync_data
+
+        :return: directory path name
+
+        :rtype: string
+        """
+        sync_target = self.mountpoint + '/@'
+        if self.custom_args['root_is_snapshot']:
+            sync_target = self.mountpoint + '/@/.snapshots/1/snapshot'
```

Comment:
Redefinitions aren't nice, as well as accessing keys for checks unsafely and incrementing immutable strings. I also find that code confusing. I would suggest to write it that way:

```python
def get_mountpoint(self) -> str:
    sync_target:Dict[str] = [self.mountpoint, '@']
    if self.custom_args.get("root_is_snapshot"):  # This will fire if key is False or is not present at all
        sync_target.extend([".snapshots", "1", "snapshot"])
    return os.path.join(*sync_target)
```

Reply:
yes this is nice and with annotations